fix_packages.php

```diff
@@ -40,80 +36,230 @@
+		$context['page_title'] = $this->lang('log_packages_title_' . (!empty($this->isInstalling) ? 'installed' : 'removed'));
 
-	// Sorry, only logged in admins...unless you want so.
-	if(empty($context['override_security']))
-		isAllowedTo('admin_forum');
+		// A nice menu button
+		cache_put_data('menu_buttons-' . implode('_', $user_info['groups']) . '-' . $user_info['language'], null);
+		add_integration_function('integrate_menu_buttons', 'FixXP::menuButton', false);
+		setupThemeContext(true);
+	}
 
-	$context['install'] = isset($_GET['uninstall']) ? 0 : 1;
+	private function buildListOptions(): array
+	{
+		global $boardurl, $context;
+
+		return [
+			'id' => 'log_packages',
+			'title' => $context['page_title'],
+			'get_items' => [
+				'function' => 'FixXP::list_getPacks',
+				'params' => [
+					$this->isInstalling
+				],
+			],
+			'get_count' => [
+				'function' => 'FixXP::list_getNumPacks',
+				'params' => [
+					$this->isInstalling
+				],
+			],
+			'columns' => [
+				'name' => [
+					'header' => [
+						'value' => $this->lang('mod_name'),
+					],
+					'data' => [
+						'db' => 'name',
+					],
+				],
+				'version' => [
+					'header' => [
+						'value' => $this->lang('mod_version'),
+					],
+					'data' => [
+						'db' => 'version',
+					],
+				],
+				'install_date' => [
+					'header' => [
+						'value' => $this->lang('mod_' . (!empty($this->isInstalling) ? 'installed' : 'removed')),
+					],
+					'data' => [
+						'function' => function($data)
+						{
+							return timeformat($data['time_' . (!empty($this->isInstalling) ? 'installed' : 'removed')]);
+						},
+					],
+				],
+				'check' => [
+					'header' => [
+						'value' => '<input type="checkbox" onclick="invertAll(this, this.form);" class="input_check" />',
+					],
+					'data' => [
+						'function' => function($data)
+						{
+							return '<input type="checkbox" name="remove[]" value="' . $data['id_install'] . '"  class="input_check" />';
+						},
+						'class' => 'centertext',
+					],
+				],
+			],
+			'form' => [
+				'href' => $boardurl . '/fix_packages.php?' . $context['session_var'] . '=' . $context['session_id'] . (!empty($this->isInstalling) ? '' : ';uninstall'),
+			],
+			'additional_rows' => [
+				[
+					'position' => 'below_table_data',
+					'value' => '
+						<a href="' . $boardurl . '/fix_packages.php' . (!empty($this->isInstalling) ? '?uninstall' : '') . '">[ ' . (!empty($this->isInstalling) ? $this->lang('uninstall') : $this->lang('install')) . ' ]</a>
+						<input type="submit" name="remove_packages" value="' . $this->lang('pack_button_' . (!empty($this->isInstalling) ? 'remove' : 'install')) . '" class="button_submit" />
+						<input type="submit" name="remove_hooks" value="' . $this->lang('remove_hooks') . '" class="button_submit" />',
+					'class' => 'righttext',
+				]
+			],
+		];
+	}
 
-	if (!empty($_POST['remove']) && is_array($_POST['remove']))
+	public static function list_getPacks(bool $isInstalling): array
```

Comment:
```suggestion
	public static function list_getPacks(int $start, int $items_per_page, string $sort, bool $isInstalling): array
```

also, why are these two methods static?

Reply:
Honestly can't remember.  I wrapped it in a class less for the object and more for ensuring the functions are local to the tool.  We can do that though if it needs done.